### PR TITLE
added visualize_node and fixed irrt* bugs

### PR DESCRIPTION
**Pull request by sushant1212**
1. Added `visualize_node` using node_list
2. Fixed I-RRT* bugs
3. Made `rrt.py` and `informed_rrtstar.py` to return an additional `info_dict` which contains a `node_list`

### CODE REVIEW

File: gennav/planners/rrt/informed_rrtstar.py [hunk 1]
```diff
@@ -228,18 +218,15 @@ def plan(self, start, goal, env):
...
-                    math.sqrt(
-                        (x_goal.state.position.x - x_new.state.position.x) ** 2
-                        + (x_goal.state.position.y - x_new.state.position.y) ** 2
-                    )
-                    < self.goal_distance
-                ):
+                goal_traj = Trajectory(path=[x_new.state, x_goal.state])
+
+                if math.sqrt(
+                    (x_goal.state.position.x - x_new.state.position.x) ** 2
```
Comment (threewisemonkeys-as) [inline gennav/planners/rrt/informed_rrtstar.py]: can you use the compute_dist function here

File: tests/test_planners/informed_rrtstar_test.py [hunk 2]
```diff
@@ -10,20 +10,19 @@ def test_informedrrtstar():
...
         neighbourhood_radius=0.5,
         goal_distance=0.5,
         max_iter=2000,
     )
-    start = RobotState(position=Point(1, 1))
-    goal = RobotState(position=Point(10, 10))
-    path = my_tree.plan(start, goal, poly)
+    start = RobotState(position=Point(0, 0))
+    goal = RobotState(position=Point(6, 6))
+    (path, _) = my_tree.plan(start, goal, poly)
```
Comment (threewisemonkeys-as) [inline tests/test_planners/informed_rrtstar_test.py]: Parantheses arent required. `path, _ = ...` should work

**Review by threewisemonkeys-as — CHANGES_REQUESTED** [review]

File: gennav/planners/rrt/informed_rrtstar.py [hunk 3]
```diff
@@ -45,16 +44,14 @@ def plan(self, start, goal, env):
 
         Returns:
             [gennav.utils.Trajectory]: The path as a Trajectory
```
Comment (threewisemonkeys-as) [inline gennav/planners/rrt/informed_rrtstar.py]:
```suggestion
            gennav.utils.Trajectory, info: The path as a Trajectory along with information dictionary
```

Comment (threewisemonkeys-as) [reply gennav/planners/rrt/informed_rrtstar.py]: Could you make this change for all the planners?

**Review by threewisemonkeys-as — CHANGES_REQUESTED** [review]

File: gennav/planners/rrt/informed_rrtstar.py (hunk 3 above)
Comment (sushant1212) [reply gennav/planners/rrt/informed_rrtstar.py]: Ok

**Review by threewisemonkeys-as — APPROVED** [review]

**Pull request merged**